src/Numerical/Spectral/DefiniteIntegral.cpp

```diff
@@ -0,0 +1,45 @@
+
+#include "Numerical/Spectral/DefiniteIntegral.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Numerical/Spectral/LegendreGaussLobatto.hpp"
+#include "Utilities/Blas.hpp"
+
+namespace {
+
+template <size_t Dim>
+DataVector integrate_over_last_dimension(const DataVector& f,
+                                         const Index<Dim>& extents) {
+  static_assert(Dim > 1, "Expect dimension to be at least 2.");
+  const size_t extents_in_last_dim = extents[Dim - 1];
+  const DataVector& w = Basis::lgl::quadrature_weights(extents_in_last_dim);
+  const size_t reduced_size =
+      extents.product() / extents_in_last_dim;  // yes, int division
```

Comment:
I think we can just say 
`const size_t reduced_size = extents.product<Dim-1>();`
since we want the product of all but the last dimension...